### PR TITLE
Improve error message around asset checksum mismatch

### PR DESCRIPTION
**Pull request by palourde**
Signed-off-by: Simon Plourde <simon@sensu.io>

## What is this change?

It fixes the error message around asset checksum mismatch.

## Why is this change necessary?

Closes https://github.com/sensu/sensu-go/issues/3674
Closes https://github.com/sensu/sensu-go/issues/4064

## Does your change need a Changelog entry?

Added
## Do you need clarification on anything?

Nope

## Were there any complications while making this change?

Nope

## Have you reviewed and updated the documentation for this change? Is new documentation required?

Nope

## How did you verify this change?

Manually verified

## Is this change a patch?

Yep

### CODE REVIEW

File: asset/boltdb_manager.go [hunk 1]
```diff
@@ -133,7 +134,16 @@ func (b *boltDBAssetManager) Get(ctx context.Context, asset *corev2.Asset) (*Run
...
 		if err := b.verifier.Verify(tmpFile, asset.Sha512); err != nil {
-			return err
+			// Attempt to retrieve the size of the downloaded asset
+			var size int64
+			if fileInfo, err := tmpFile.Stat(); err == nil {
+				size = fileInfo.Size() / 1000 // kilobytes
+			}
+
+			return fmt.Errorf(
+				"could not validate downloaded asset %q (%d kilobytes): %s",
```
Comment (echlebek) [inline asset/boltdb_manager.go:144]: Maybe a good use case for https://godoc.org/github.com/dustin/go-humanize#Bytes ?

File: asset/verifier.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ func (v *Sha512Verifier) Verify(rs io.ReadSeeker, desiredSHA string) error {
 	}
 
 	if foundSHA := hex.EncodeToString(h.Sum(nil)); foundSHA != desiredSHA {
-		return fmt.Errorf("sha512 '%s' does not match specified sha512: '%s'", desiredSHA, foundSHA)
+		return fmt.Errorf("sha512 of downloaded asset (%s) does not match specified sha512 in asset definition (%s)", foundSHA, desiredSHA)
 	}
```
Comment (echlebek) [inline asset/verifier.go:37]: Since you're working on this section of code already, do you mind fixing https://github.com/sensu/sensu-go/issues/4064 ?

**Review by echlebek — COMMENTED** [review]
Great work! Wondering if we can get two birds stoned at once with this PR.

File: asset/boltdb_manager.go (hunk 1 above)
Comment (palourde) [reply asset/boltdb_manager.go:144]: Updated with this lib, much better 👌 
```
Output:    error getting assets for check: could not validate downloaded asset "sensu-go-hello-world" (5.8 MB): sha512 of downloaded asset (8b43b938264a96b236fbec55814d34745cd695b092a9806e21f58f041d8a00a56e8aeadb1bdeb26e0612b332b51c998d3273906e045add4c14c5173f49b2aa3b) does not match specified sha512 in asset definition (06665fea5b7c7ee15e7822820aa7ddb791cc6338e38444e976e601bc7d7970592e806a7b88733690a238b7325437d31f85e98ae2fe47b008ca09c86530da9600)
```

File: asset/verifier.go (hunk 2 above)
Comment (palourde) [reply asset/verifier.go]: Just pushed https://github.com/sensu/sensu-go/pull/4118/commits/0f5af5329833fc6500a2b8c974609e3b54a2b8cd

**Review by echlebek — APPROVED** [review]
:+1:

**Pull request merged**